src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1171,21 +1172,28 @@ private void addCommentTags(Element element, List<? extends DocTree> tags, boole
         }
     }
 
-    boolean ignoreNonInlineTag(DocTree dtree) {
+    boolean ignoreNonInlineTag(DocTree dtree, List<Name> openTags) {
         Name name = null;
-        if (dtree.getKind() == Kind.START_ELEMENT) {
-            StartElementTree setree = (StartElementTree)dtree;
-            name = setree.getName();
-        } else if (dtree.getKind() == Kind.END_ELEMENT) {
-            EndElementTree eetree = (EndElementTree)dtree;
-            name = eetree.getName();
+        Kind kind = dtree.getKind();
+        if (kind == Kind.START_ELEMENT) {
+            name = ((StartElementTree)dtree).getName();
+        } else if (kind == Kind.END_ELEMENT) {
+            name = ((EndElementTree)dtree).getName();
         }
 
         if (name != null) {
             HtmlTag htmlTag = HtmlTag.get(name);
```

Comment:
minor P5 suggestion to consider in future:
maybe add convenience forms for `HtmlTag.of` that accept a `DocTree`.

```
HtmlTree htmlTag = HtmlTree.of(tree);
```

```
HtmlTree of(DocTree dtree) {
    return switch (tree.getKind()) {
        case START_ELEMENT -> of(((StartElementTree)dtree).getName());
        case END_ELEMENT -> of(((StartElementTree)dtree).getName());
        default -> throw new IllegalArgumentException(tree.getKind().toString());
    }
}
```